js/main.js

```diff
@@ -53,16 +45,18 @@ function saveEntry(event) {
     $noEntry.className = 'hidden';
     $form.reset();
   } else {
+    data.editing.title = $form.elements.title.value;
+    data.editing.photo = $form.elements.photo.value;
+    data.editing.notes = $form.elements.notes.value;
     for (var i = 0; i < data.entries.length; i++) {
       if (data.entries[i].entryId === data.editing.entryId) {
-        data.editing.title = $form.elements.title.value;
-        data.editing.photo = $form.elements.photo.value;
-        data.editing.notes = $form.elements.notes.value;
+        data.entries[i] = data.editing;
+        data.entries.splice(i, 1, dataObject);
```

Comment:
I don't think this line is necessary, you already updating the `data.editing` object with the new values, the line above this one is already replacing the value for you.